packages/select/src/MenuButton.tsx

```diff
@@ -0,0 +1,290 @@
+import React, { useCallback, useContext, useState } from 'react';
+import { cx, css } from '@leafygreen-ui/emotion';
+import CaretDownIcon from '@leafygreen-ui/icon/dist/CaretDown';
+import { useUsingKeyboardContext } from '@leafygreen-ui/leafygreen-provider';
+import { breakpoints } from '@leafygreen-ui/tokens';
+import { colorSets, mobileSizeSet, Mode, sizeSets } from './styleSets';
+import SelectContext from './SelectContext';
+
+const menuButtonStyle = css`
+  cursor: pointer;
+  outline: none;
+  border: 1px solid;
+  border-radius: 3px;
+  font-weight: initial;
+  user-select: none;
+  margin-top: 2px;
+`;
+
+const menuButtonContentsStyle = css`
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  height: 100%;
+`;
+
+const menuButtonTextStyle = css`
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+`;
+
+type Props = {
+  children: React.ReactNode;
+  value: string;
+  text: React.ReactNode;
+  name: string;
+  deselected: boolean;
+  readOnly?: boolean;
+  refElement: React.MutableRefObject<HTMLDivElement | null>;
+  onFocusFirstOption: () => void;
+  onFocusLastOption: () => void;
+  onDeselect: () => void;
+  onClose: () => void;
+  onOpen: () => void;
+} & Required<
+  Pick<
+    JSX.IntrinsicElements['div'],
```

Comment:
Know we mentioned that this type includes `ref`, but don't see ref included in the props. Are we fine with leaving as-is? Or were we going to establish a new pattern around this? Wondering because I'm going to try and standardize across the repo

Reply:
we're only picking the aria props from it here so whether or not it has ref doesn't change the outcome :)

Reply:
👍